### ISSUE TITLE
Specified artifacts not showing on artifacts directory for local build

### ISSUE THREAD

**Issue by davifelix5** [expo/eas-cli#2522]
### Summary

When specifying build artifacts path on `eas.json` and setting` EAS_LOCAL_BUILD_ARTIFACTS_DIR`, the `.dSYM.zip` file do not appear on the specified directory when running `eas build -p ios --local`. The expected behavior is for the `.dSYM.zip` file and the `generated` folder to appear alongside the `.ipa` artifact, just as it happens with the downloaded artifacts on a remote build.

### Managed or bare?

bare

### Environment

expo-doctor:
```
Detailed check results:

The following packages should be updated for best compatibility with the installed expo version:
  expo@51.0.26 - expected version: ~51.0.31
  expo-dev-client@4.0.22 - expected version: ~4.0.25
  expo-image@1.12.13 - expected version: ~1.12.15
  expo-router@3.5.21 - expected version: ~3.5.23
  expo-updates@0.25.22 - expected version: ~0.25.24
  react-native-reanimated@3.15.0 - expected version: ~3.10.1
  jest-expo@51.0.3 - expected version: ~51.0.4
Your project may not work correctly until you install the expected versions of the packages.
Found outdated dependencies
Advice: Use 'npx expo install --check' to review and upgrade your dependencies.

One or more checks failed, indicating possible issues with the project.
```

expo-env-info:
```
  expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.5
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.15.1 - ~/.nvm/versions/node/v20.15.1/bin/node
      npm: 10.7.0 - ~/.nvm/versions/node/v20.15.1/bin/npm
      Watchman: 2024.04.29.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.15.2 - /Users/myuser/.rvm/gems/ruby-3.0.4/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.5, iOS 17.5, macOS 14.5, tvOS 17.5, visionOS 1.2, watchOS 10.5
    IDEs:
      Android Studio: 2023.3 AI-233.14808.21.2331.11709847
      Xcode: 15.4/15F31d - /usr/bin/xcodebuild
    npmPackages:
      expo: ~51.0.26 => 51.0.26 
      expo-router: ~3.5.21 => 3.5.21 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.74.5 => 0.74.5 
      react-native-web: ~0.19.10 => 0.19.11 
    npmGlobalPackages:
      eas-cli: 11.0.2
    Expo Workflow: bare
```

### Reproducible demo or steps to reproduce from a blank project

1. Create a expo project
2. Install `eas`
3. Create a `eas.json` file

```json
{
    "build": {
        "production": {
            "ios": {
              "buildArtifactPaths": ["ios/build/*"]
            }
         }
     }
}

```

1. Run `EAS_LOCAL_BUILD_ARTIFACTS_DIR=$(pwd)/build eas build -p ios --local` on the root of the project
2. See that the the `.dSYM.zip` does not appear when running `ls build`.

**Comment by szdziedzic** [comment]
https://github.com/expo/eas-build/pull/439

**Comment by szdziedzic** [comment]
It should be released as a part of 12.0.0

**Comment by davifelix5** [comment]
> It should be released as a part of 12.0.0

Thank you for the fast reply!
